src/v1/Controllers/Item.php

```diff
@@ -348,9 +349,50 @@ public function postItem(Request $request, Response $response, $args): Response
     // Validate the data format
     $dataFormat = [
       'name'    => 'required|type:string',
-      'type_id' => 'required|type:integer|integer'
+      'type_id' => 'required|type:integer|integer',
+      'parent_id' => 'type:integer|integer'
     ];
     \App\v1\Common::validateData($data, $dataFormat);
+    // Checks about tree type
+    $type = \App\v1\Models\Config\Type::find($data->type_id);
+    if ($type->tree)
+    {
+      // in case of type is a tree, check if parent_id exists
+      if (property_exists($data, 'parent_id'))
+      {
+        $parentItem = \App\v1\Models\Item::find($data->parent_id);
+        if (is_null($parentItem))
+        {
+          throw new \Exception("The parent item has not be found", 400);
+        }
+        // Check now if the parent_id is the same type_id
+        if ($parentItem->type_id != $data->type_id)
+        {
+          throw new \Exception("The parent item has not the same type", 400);
+        }
+      }
+      else {
+        if (!$type->allowtreemultipleroots)
+        {
+          // check if have yet a root item
+          $otherItemsOfTree = \App\v1\Models\Item::where('type_id', $data->type_id)
+            ->take(1)
+            ->first();
+          if (!is_null($otherItemsOfTree))
+          {
+            throw new \Exception("This type can only have one root item", 400);
+          }
+        }
+      }
+    }
+    else {
+      // in case of type is not a tree, check if parent_id is not set
+      if (property_exists($data, 'parent_id'))
```

Comment:
```suggestion
    elseif (property_exists($data, 'parent_id'))
    {
      // in case of type is not a tree, check if parent_id is not set
```

(nitpicking)